Wikipedia/UI-V5/WMFArticleListCollectionViewController.m

```diff
@@ -126,10 +128,34 @@ - (NSString*)debugDescription {
     return [NSString stringWithFormat:@"%@ dataSourceClass: %@", self, [self.dataSource class]];
 }
 
-- (void)refreshVisibleCells {
-    [self.collectionView wmf_enumerateVisibleCellsUsingBlock:
-     ^(WMFArticlePreviewCell* cell, NSIndexPath* path, BOOL* _) {
+#pragma mark - Stay Fresh... yo
+
+- (void)observeArticleUpdates {
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:MWKArticleSavedNotification object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(articleUpdatedWithNotification:) name:MWKArticleSavedNotification object:nil];
+}
+
+- (void)unobserveArticleUpdates {
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:MWKArticleSavedNotification object:nil];
+}
+
+- (void)articleUpdatedWithNotification:(NSNotification*)note {
+    MWKArticle* article = note.userInfo[MWKArticleKey];
+    [self refreshAnyVisibleCellsWhichAreShowingTitle:article.title];
+}
+
+- (void)refreshAnyVisibleCellsWhichAreShowingTitle:(MWKTitle*)title {
+    NSArray* indexPathsToRefresh = [[self.collectionView indexPathsForVisibleItems] bk_select:^BOOL (NSIndexPath* path) {
+        WMFArticlePreviewCell* cell = (WMFArticlePreviewCell*)[self.collectionView cellForItemAtIndexPath:path];
```

Comment:
convention for retrieving model objects for an index path is to access them directly, or via any dataSource or other controller, as opposed to its corresponding cell. in general we should only try to access the information we need in a given method, and avoid views if possible. (hint: watch Andy's talk about the "value layer game" :wink:)